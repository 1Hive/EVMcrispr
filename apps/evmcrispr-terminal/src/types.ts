export type BareScript = {
  title: string;
  script: string;
};

export type StoredScript = {
  title: string;
  date: Date;
  script: string;
};
