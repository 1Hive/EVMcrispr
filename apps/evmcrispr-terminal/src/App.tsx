import "@fontsource/ubuntu-mono";
import {
  Navigate,
  Route,
  RouterProvider,
  createHashRouter,
  createRoutesFromElements,
} from "react-router-dom";
import { ChakraProvider, DarkMode, extendTheme } from "@chakra-ui/react";

import theme from "./theme";
import Wagmi from "./providers/Wagmi";

import Landing from "./pages/Landing";
import Terminal from "./pages/Terminal";
import Fonts from "./theme/Fonts";

const App = () => {
  const router = createHashRouter(
    createRoutesFromElements(
      <>
        <Route index element={<Landing />} />
        <Route path="terminal" element={<Terminal />}>
          <Route path=":scriptId" element={<Terminal />} />
        </Route>
        <Route path="*" element={<Navigate to="/" replace />} />
      </>,
    ),
  );
  return (
    <div className="App">
      <ChakraProvider theme={extendTheme(theme)}>
        <Fonts />
        <DarkMode>
          <Wagmi>
            <RouterProvider router={router} />
          </Wagmi>
        </DarkMode>
      </ChakraProvider>
    </div>
  );
};

export default App;
