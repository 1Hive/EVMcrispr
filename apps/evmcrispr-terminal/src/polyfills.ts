import Buffer from "buffer";

declare global {
  interface Window {
    Buffer: any;
  }
}

// polyfill Buffer for client
if (!window.Buffer) {
  window.Buffer = Buffer;
}
