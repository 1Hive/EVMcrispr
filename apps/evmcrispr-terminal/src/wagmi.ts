import { createConfig, http } from "wagmi";
import { injected, walletConnect } from "wagmi/connectors";

import * as _chains from "wagmi/chains";
import type { Chain, Transport } from "viem";

import { safe } from "./overrides/safe";

const WALLETCONNECT_PROJECT_ID = import.meta.env.VITE_WALLETCONNECT_PROJECT_ID;
const ALCHEMY_API_KEY = import.meta.env.VITE_ALCHEMY_API_KEY;

const isIframe = window.self !== window.top;

const alchemyTransports = ALCHEMY_API_KEY && {
  [_chains.mainnet.id]:
    `https://eth-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.sepolia.id]:
    `https://eth-sepolia.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.polygon.id]:
    `https://polygon-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.polygonAmoy.id]:
    `https://polygon-amoy.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.polygonZkEvm.id]:
    `https://polygonzkevm-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.polygonZkEvmCardona.id]:
    `https://polygonzkevm-cardona.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.optimism.id]:
    `https://opt-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.optimismSepolia.id]:
    `https://opt-sepolia.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.arbitrum.id]:
    `https://arb-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.arbitrumSepolia.id]:
    `https://arb-sepolia.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.base.id]: `https://base-mainnet.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
  [_chains.baseSepolia.id]:
    `https://base-sepolia.g.alchemy.com/v2/${ALCHEMY_API_KEY}`,
};
const chains = Object.values(_chains) as unknown as [Chain, ...Chain[]];
export const transports = chains.reduce(
  (acc, { id }) => {
    acc[id] = alchemyTransports?.[id] ? http(alchemyTransports[id]) : http();
    return acc;
  },
  {} as Record<number, Transport>,
);

export const config = createConfig({
  chains,
  connectors: [
    !isIframe && injected(),
    !isIframe &&
      WALLETCONNECT_PROJECT_ID &&
      walletConnect({
        projectId: WALLETCONNECT_PROJECT_ID,
      }),
    isIframe &&
      safe({
        allowedDomains: [/app.safe.global$/],
        unstable_getInfoTimeout: 500,
      }),
  ].filter(Boolean),
  transports,
});
