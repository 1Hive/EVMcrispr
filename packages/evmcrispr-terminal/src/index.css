body,
html,
#root {
  height: 100%;
  background: black;
  margin: 0;
  padding: 0;
}

.App {
  height: 100%;
  /* remove this line to see div.app is no more 100% height */
  background: black;
  color: white;
}

/* The typing effect */
@keyframes typing {
  from {
    width: 0;
    border-color: #fff;
  }

  to {
    width: 16.5rem;
  }

  99% {
    border-color: #fff;
  }

  100% {
    border-color: transparent;
  }
}

.box {
  position: relative;
  z-index: 0;
  background: rgba(150, 238, 100, 0.05);
  clip-path: polygon(0 60.00px, 30px 30px, 60.00px 0, calc(100% - 60.00px) 0, calc(100% - 30px) 30px, 100% 60.00px, 100% calc(100% - 60.00px), calc(100% - 30px) calc(100% - 30px), calc(100% - 60.00px) 100%, 60.00px 100%, 30px calc(100% - 30px), 0 calc(100% - 60.00px));

}

.box:before {
  content: "";
  position: absolute;
  inset: 0;
  z-index: -1;
  background: #75f248;
  clip-path: polygon(0 60.00px, 30px 30px, 60.00px 0, calc(100% - 60.00px) 0, calc(100% - 30px) 30px, 100% 60.00px, 100% calc(100% - 60.00px), calc(100% - 30px) calc(100% - 30px), calc(100% - 60.00px) 100%, 60.00px 100%, 30px calc(100% - 30px), 0 calc(100% - 60.00px), 0 60.00px, 2px calc(60.00px + 0.83px), 2px calc(100% - 60.00px - 0.83px), calc(30px + 1.41px) calc(100% - 30px - 1.41px), calc(60.00px + 0.83px) calc(100% - 2px), calc(100% - 60.00px - 0.83px) calc(100% - 2px), calc(100% - 30px - 1.41px) calc(100% - 30px - 1.41px), calc(100% - 2px) calc(100% - 60.00px - 0.83px), calc(100% - 2px) calc(60.00px + 0.83px), calc(100% - 30px - 1.41px) calc(30px + 1.41px), calc(100% - 60.00px - 0.83px) 2px, calc(60.00px + 0.83px) 2px, calc(30px + 1.41px) calc(30px + 1.41px), 2px calc(60.00px + 0.83px));
}