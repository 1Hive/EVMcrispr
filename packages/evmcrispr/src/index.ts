export { default as Connector } from './Connector';
export { default as EVMcrispr } from './EVMcrispr';
export { default as evmcl } from './evmcl';
export {
  arg,
  blockNumber,
  timestamp,
  oracle,
  not,
  and,
  or,
  xor,
  iif,
  paramValue,
} from './utils/acl';
export { normalizeActions } from './utils/normalizers';
export { encodeActCall, encodeCallScript } from './utils/evmscripts';
export { ErrorException, ErrorInvalid, ErrorNotFound } from './errors';
export type { ErrorOptions } from './errors';
export type {
  Address,
  Action,
  ActionFunction,
  App,
  AppCache,
  AppIdentifier,
  CallScriptAction,
  CompletePermission,
  Entity,
  EVMcrisprOptions,
  ForwardOptions,
  LabeledAppIdentifier,
  ParsedApp,
  Params,
  Permission,
  PermissionP,
  PermissionMap,
  Repo,
  Role,
  RoleHash,
} from './types';
