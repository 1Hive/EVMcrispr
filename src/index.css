body { background: black; color: white }
body, h1, input {font-weight: bold; font-size: 40px; font-family: Monaco, Menlo, "Ubuntu Mono", Consolas, source-code-pro, monospace;}
h1 {color: #D8FA3C; }
input[type=button] {background-color: #D8FA3C;cursor:pointer; margin-left: 10px}
