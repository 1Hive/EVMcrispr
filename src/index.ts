export { default as Connector } from "./Connector";
export { default as EVMcrispr } from "./EVMcrispr";
export { default as evmcl } from "./evmcl";
export { arg, blockNumber, timestamp, oracle, not, and, or, xor, iif, paramValue } from "./helpers/acl";
export { normalizeActions } from "./helpers/normalizers";
export { encodeActCall, encodeCallScript } from "./helpers/evmscripts";
export { ErrorException, ErrorInvalid, ErrorOptions, ErrorNotFound } from "./errors";
export {
  Address,
  Action,
  ActionFunction,
  ActionInterpreter,
  App,
  AppCache,
  AppIdentifier,
  CompletePermission,
  Entity,
  EVMcrisprOptions,
  ForwardOptions,
  LabeledAppIdentifier,
  Permission,
  PermissionP,
  Repo,
} from "./types";
