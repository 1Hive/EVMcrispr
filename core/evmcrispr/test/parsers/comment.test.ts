import { runParser } from '@1hive/evmcrispr-test-common';
import type { Case } from '@1hive/evmcrispr-test-common';

import { scriptParser } from '../../src';

describe('Parsers - comment', () => {
  describe.concurrent.each<Case>([
    {
      title: 'normal',
      value: `
    # a comment here
    load aragonos as ar

    #another one here
    set $var1 1e18

    #one at the end
  `,
    },
    {
      title: 'inline',
      value: `
      load aragonos as ar # this is an inline comment
      set $var1 1e18 #another one
    `,
    },
  ])('', ({ title, value }) => {
    it(`should parse a ${title} comment correctly`, () => {
      const parsedValue = runParser(scriptParser, value);

      expect(parsedValue).toMatchSnapshot();
    });
  });
});
