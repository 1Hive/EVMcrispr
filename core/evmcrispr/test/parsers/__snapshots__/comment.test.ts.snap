// Vitest Snapshot v1

exports[`Parsers - comment > should parse a inline comment correctly 1`] = `
Cas11AST {
  "body": [
    {
      "args": [
        {
          "left": {
            "loc": {
              "end": {
                "col": 19,
                "line": 2,
              },
              "start": {
                "col": 11,
                "line": 2,
              },
            },
            "type": "ProbableIdentifier",
            "value": "aragonos",
          },
          "loc": {
            "end": {
              "col": 25,
              "line": 2,
            },
            "start": {
              "col": 11,
              "line": 2,
            },
          },
          "right": {
            "loc": {
              "end": {
                "col": 25,
                "line": 2,
              },
              "start": {
                "col": 23,
                "line": 2,
              },
            },
            "type": "ProbableIdentifier",
            "value": "ar",
          },
          "type": "AsExpression",
        },
      ],
      "loc": {
        "end": {
          "col": 25,
          "line": 2,
        },
        "start": {
          "col": 6,
          "line": 2,
        },
      },
      "name": "load",
      "opts": [],
      "type": "CommandExpression",
    },
    {
      "args": [
        {
          "loc": {
            "end": {
              "col": 15,
              "line": 3,
            },
            "start": {
              "col": 10,
              "line": 3,
            },
          },
          "type": "VariableIdentifier",
          "value": "$var1",
        },
        {
          "loc": {
            "end": {
              "col": 20,
              "line": 3,
            },
            "start": {
              "col": 16,
              "line": 3,
            },
          },
          "power": 18,
          "type": "NumberLiteral",
          "value": "1",
        },
      ],
      "loc": {
        "end": {
          "col": 20,
          "line": 3,
        },
        "start": {
          "col": 6,
          "line": 3,
        },
      },
      "name": "set",
      "opts": [],
      "type": "CommandExpression",
    },
  ],
  "type": "Program",
}
`;

exports[`Parsers - comment > should parse a normal comment correctly 1`] = `
Cas11AST {
  "body": [
    {
      "args": [
        {
          "left": {
            "loc": {
              "end": {
                "col": 17,
                "line": 3,
              },
              "start": {
                "col": 9,
                "line": 3,
              },
            },
            "type": "ProbableIdentifier",
            "value": "aragonos",
          },
          "loc": {
            "end": {
              "col": 23,
              "line": 3,
            },
            "start": {
              "col": 9,
              "line": 3,
            },
          },
          "right": {
            "loc": {
              "end": {
                "col": 23,
                "line": 3,
              },
              "start": {
                "col": 21,
                "line": 3,
              },
            },
            "type": "ProbableIdentifier",
            "value": "ar",
          },
          "type": "AsExpression",
        },
      ],
      "loc": {
        "end": {
          "col": 23,
          "line": 3,
        },
        "start": {
          "col": 4,
          "line": 3,
        },
      },
      "name": "load",
      "opts": [],
      "type": "CommandExpression",
    },
    {
      "args": [
        {
          "loc": {
            "end": {
              "col": 13,
              "line": 6,
            },
            "start": {
              "col": 8,
              "line": 6,
            },
          },
          "type": "VariableIdentifier",
          "value": "$var1",
        },
        {
          "loc": {
            "end": {
              "col": 18,
              "line": 6,
            },
            "start": {
              "col": 14,
              "line": 6,
            },
          },
          "power": 18,
          "type": "NumberLiteral",
          "value": "1",
        },
      ],
      "loc": {
        "end": {
          "col": 18,
          "line": 6,
        },
        "start": {
          "col": 4,
          "line": 6,
        },
      },
      "name": "set",
      "opts": [],
      "type": "CommandExpression",
    },
  ],
  "type": "Program",
}
`;
